crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/generated_acir.rs

```diff
@@ -773,12 +680,13 @@ impl GeneratedAcir {
         // - 2^{max_bits} - k == q * 2^{max_bits} + r
         // - This is only the case when q == 0 and r == 2^{max_bits} - k
         //
-        let mut expr = Expression::default();
-        expr.push_addition_term(two_max_bits, q_witness);
-        expr.push_addition_term(FieldElement::one(), r_witness);
-        self.push_opcode(AcirOpcode::Arithmetic(&comparison_evaluation - &expr));
-
-        Ok(q_witness)
+        let (q, _) = self.euclidean_division(
+            &comparison_evaluation,
+            &Expression::from(two_max_bits),
+            max_bits + 1,
```

Comment:
Can you add a comment on why it is max_bits + 1?

Reply:
Done